### PR TITLE
migrate keytar to use n-api

### PR DESCRIPTION
**Pull request by jkleinsc**
This PR migrates keytar from nan to [N-API](https://nodejs.org/dist/latest-v14.x/docs/api/n-api.html#n_api_n_api) via the [node-addon-api module](https://github.com/nodejs/node-addon-api).

Migrating to N-API provides ABI stability meaning that versions of keytar compiled for one major version can run on later major versions of Node.js without recompilation.  This means that issues like #174 would be non-issues in the future.

### CODE REVIEW

**Comment by shiftkey** [comment]
This has been published to npm as [`v6.0.0-beta.0`](https://www.npmjs.com/package/keytar/v/6.0.0-beta.0) and also as the `beta` label if anyone wants to test these changes.

File: src/main.cc [hunk 1]
```diff
@@ -1,131 +1,139 @@
...
 }
 
-NAN_METHOD(DeletePassword) {
-  if (!info[0]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'service' must be a string");
-    return;
+Napi::Value DeletePassword(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  if (!info[0].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'service' must be a string");
```
Comment (vadim-termius) [inline src/main.cc:72]: Is `ThrowAsJavaScriptException` missing here?

Comment (vadim-termius) [reply src/main.cc:72]: I think it makes sense to extract a method like that:

```
void ThrowTypeError(napi_env env, const char* message) {
    Napi::TypeError::New(env, message).ThrowAsJavaScriptException();
}
```

Comment (jkleinsc) [reply src/main.cc:72]: @vadim-termius, you are right -- I missed this in the conversion.  https://github.com/atom/node-keytar/pull/270 will resolve this.

**Comment by vadim-termius** [comment]
> Migrating to N-API provides ABI stability meaning that versions of keytar compiled for one major version can run on later major versions of Node.js without recompilation. This means that issues like #174 would be non-issues in the future.

Just saw the 6th release and remembered my latest research on this topic. According to the [documentation](https://nodejs.org/dist/latest-v14.x/docs/api/n-api.html#n_api_implications_of_abi_stability) there is no guarantee in ABI stability if the module uses API from `uv.h`. I see there are a few includes of this header but I didn't find usage of the related API. I think, if it is possible, it is better to remove them.

**Comment by shiftkey** [comment]
@vadim-termius nice spot - I couldn't spot anything obviously dependent on `uv.h` locally so I've opened #278 to see if it's feasible to include in an update.

**Pull request merged**